src/types/timestamp.rs

```diff
@@ -572,6 +572,66 @@ impl Timestamp {
     pub fn precision(&self) -> Precision {
         self.precision
     }
+
+    /// Returns the year that has been specified in the [Timestamp].
+    pub fn year(&self) -> i32 {
+        self.date_time.year()
+    }
+
+    /// Returns the month that has been specified in the [Timestamp].
+    pub fn month(&self) -> u32 {
+        self.date_time.month()
+    }
+
+    /// Returns the day that has been specified in the [Timestamp].
+    pub fn day(&self) -> u32 {
+        self.date_time.day()
+    }
+
+    /// Returns the hour(s) that has been specified in the [Timestamp].
+    pub fn hour(&self) -> u32 {
+        // verify if the timestamp has an offset
+        if let Some(offset) = self.offset {
+            // `NativeDateTime#hours()` returns hours normalized as per UTC
+            // for local time we need to +/- the difference
+            return (self.date_time.hour() as i32 + (offset.local_minus_utc() / 3600)) as u32;
```

Comment:
This looks ok to me, but if any of the unit tests we add show that something more complex is required, we can get the `DateTime<FixedOffset>` version of `self.date_time` using [`DateTime::from_local`](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.from_local) and then ask it for the corresponding field.